lib/simp/cli/commands/bootstrap.rb

```diff
@@ -155,91 +91,133 @@ def self.run(args = [])
 
     # Print intro
     system('clear')
-    puts
-    puts "*** Starting SIMP Bootstrap ***"
-    puts "   If this runs quickly, something wrong happened. To debug the problem,"
-    puts "   run 'puppet agent --test' by hand or read the log. The log can be found"
-    puts "   at '#{@logfile.path}'."
-    puts
+    say ("<%= color('=== Starting SIMP Bootstrap ===', :yellow) %>")
+
+    # Set an interrupt trap if safe mode is enabled
+    say "> The log can be found at '#{@logfile.path}'\n"
+    if not @unsafe
+      Signal.trap("INT") { say "<%= color('Safe mode enabled, ignoring interrupt', :magenta) %>"}
```

Comment:
* What about other signals such as SIGHUP, SIGUSR1, SIGUSR2
* I was expecting the signal trapping to be done around execs such as system() calls, not the entire bootstrap process.  In other words,  based on what Trevor said, I thought the intent was to prevent puppet calls from being interrupted.  As currently written, you can't interrupt bootstrap at all.